src/persist-client/src/internal/compact.rs

```diff
@@ -458,6 +438,50 @@ where
         }
     }
 
+    pub async fn apply(
+        res: FueledMergeRes<T>,
+        metrics: Arc<Metrics>,
+        machine: Machine<K, V, T, D>,
```

Comment:
The machine and metrics could be taken by reference... saving some unneeded clones.